src/nvim/edit.c

```diff
@@ -1532,7 +1532,10 @@ static void init_prompt(int cmdchar_todo)
 {
   char *prompt = prompt_text();
 
-  if (curwin->w_cursor.lnum < curbuf->b_prompt_start.mark.lnum) {
+  if (curwin->w_cursor.lnum < curbuf->b_prompt_start.mark.lnum
+      || (cmdchar_todo != 'O'
```

Comment:
Not feeling great about special-casing things like this. But at least we have test coverage, so can iterate.

Reply:
No `o` doesn't need special handling here, because with `o` we just create a new line below current-prompt. `O` is basically an edge case, as in the case of `O`, we replace the current-prompt line.